mantidproject/mantid
# Complain if underlying data changes in the phase quad gui

The frequency domain analysis would use incorrect workspaces in its calculations if the data in muon analysis changed

**To test:**
Open muon analysis
Load some data
Open frequency domain analysis
Change workspace to PhaseQuad and run it (you may have to change the endX)
See that the name now includes the run name (instrument + run number)
Change the method to maxent
Tick all of the output options
Run maxent
See that the output names now include which muon analysis they worked on (useful for multi period data such as HIFI 84447)

Change the data in Muon analysis (without closing the frequency GUI)
Go back to the frequency GUI
Run maxEnt - it should give an error then update the GUI
Try to run it again and it should add the output to the new group ws with correct names
Go back to FFT and try another phaseQuad - it should give an error and update
Try to run it again and it should work (add the output to the new ws and with correct names)



Fixes #22031. 

**Release Notes** 


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
